airflow/models/dag.py

```diff
@@ -2347,6 +2352,19 @@ def create_dagrun(
             creating_job_id=creating_job_id,
             data_interval=data_interval,
         )
+
+        # prevent triggering manually for scheduled DAGs, while 'allow_manual' = False
+        if (
+            run.run_type == DagRunType.MANUAL
+            and self.schedule_interval is not None
+            and self.schedule_interval != "@once"
```

Comment:
Easier to use `self.timetable.periodic`

Reply:
But currently `allow_manual=False` means “do not allow manual runs _unless the schedule is not periodic_”, which is far from obvious. If this is the semantic you are looking for, the argument should likely be renamed (something like `periodic_only`). If you want to use `allow_manual`, the semantic should be to disable all manual runs without exceptions.

Reply:
good call @uranusjr , updated. FYI @XLPI .